### PR TITLE
Ability to format user and team bio 

### LINKED ISSUE #6204: Multiline 'about' field for profile page
### Is your proposal related to a problem?

The about field currently only allows a single line of text, to ensure additional information is readable, I'd like this field to support multiple lines.

The hint text suggests 'a few sentences', but this becomes quickly unreadable.

### Describe the solution you'd like

Using - or | as separators between sentences to differentiate between points in the about section.

image

image

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds ability to format user and team bio (bold, italic, link and multi line) by using our already existing `<Editor>` component. 

User bio: 
Screenshot 2023-01-19 at 09 37 09

Screenshot 2023-01-19 at 09 26 01

Team bio:

Screenshot 2023-01-19 at 09 26 59

Screenshot 2023-01-19 at 09 27 31

Fixes #6204 

**Environment**: Staging(main branch)

### CODE REVIEW

File: packages/lib/getInnerText.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const getInnerText = (node: string | JSX.Element | JSX.Element[]): string => {
```
Comment (CarinaWolli) [inline packages/lib/getInnerText.tsx:1]: helper function to get text without html tags (used to see if bio is empty)

File: apps/web/components/getting-started/steps-views/UserProfile.tsx [hunk 2]
```diff
@@ -21,13 +21,13 @@ const UserProfile = (props: IUserProfileProps) => {
   const { user } = props;
   const { t } = useLocale();
   const avatarRef = useRef<HTMLInputElement>(null!);
-  const bioRef = useRef<HTMLTextAreaElement>(null);
   const {
-    register,
     setValue,
     handleSubmit,
+    getValues,
     formState: { errors },
```
Comment (zomars) [inline apps/web/components/getting-started/steps-views/UserProfile.tsx:28]:
```suggestion
```

File: apps/web/components/team/screens/Team.tsx [hunk 3]
```diff
@@ -12,6 +14,7 @@ type MemberType = MembersType[number];
 
 const Member = ({ member, teamName }: { member: MemberType; teamName: string | null }) => {
   const { t } = useLocale();
+  const parsedBio = parse(member.bio || "");
```
Comment (zomars) [inline apps/web/components/team/screens/Team.tsx:17]: Are we saving this data as HTML or Markdown? And if not markdown, why?

File: apps/web/pages/settings/my-account/profile.tsx [hunk 4]
```diff
@@ -46,8 +47,8 @@ const SkeletonLoader = ({ title, description }: { title: string; description: st
       <Meta title={title} description={description} />
       <div className="mt-6 mb-8 space-y-6 divide-y">
         <div className="flex items-center">
-          <SkeletonAvatar className=" h-12 w-12 px-4" />
-          <SkeletonButton className=" h-6 w-32 rounded-md p-5" />
+          <SkeletonAvatar className="h-12 w-12 px-4 " />
+          <SkeletonButton className="h-6 w-32 rounded-md p-5 " />
```
Comment (zomars) [inline apps/web/pages/settings/my-account/profile.tsx:51]:
```suggestion
          <SkeletonAvatar className="h-12 w-12 px-4" />
          <SkeletonButton className="h-6 w-32 rounded-md p-5" />
```

File: apps/web/package.json [hunk 5]
```diff
@@ -71,6 +71,7 @@
     "googleapis": "^84.0.0",
     "gray-matter": "^4.0.3",
     "handlebars": "^4.7.7",
+    "html-react-parser": "^3.0.7",
```
Comment (zomars) [inline apps/web/package.json:74]: Can't we use `markdown-it` for this?

**Review by zomars — COMMENTED** [review]
Left some comments

File: packages/ui/components/editor/Editor.tsx [hunk 6]
```diff
@@ -63,7 +66,7 @@ export const Editor = (props: TextEditorProps) => {
             excludedToolbarItems={props.excludedToolbarItems}
             variables={props.variables}
           />
-          <div className="editor-inner">
+          <div className={`editor-inner ${props.height}`}>
```
Comment (sean-brydon) [inline packages/ui/components/editor/Editor.tsx:69]: Please pass this as inside a style tag - theres no way to know if the css for the entered prop is generated in our tailwind file (Or allow string to be a union type of things we know will be generated)

Comment (zomars) [reply packages/ui/components/editor/Editor.tsx:69]: good point

File: apps/web/components/team/screens/Team.tsx (hunk 3 above)
Comment (CarinaWolli) [reply apps/web/components/team/screens/Team.tsx:17]: It is saved as HTML. The `Editor` component creates a HTML with styles from the` stylesEditor.css` file.  Should I convert the bio to markdown before saving it?

File: apps/web/package.json (hunk 5 above)
Comment (PeerRich) [reply apps/web/package.json:74]: @CarinaWolli

**Comment by PeerRich** [comment]
CleanShot 2023-01-21 at 18 20 05@2x
LGTM after merge conflicts

File: packages/lib/turndownService.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import TurndownService from "turndown";
+
+const turndownService = new TurndownService();
+
+turndownService.addRule("newLine", {
+  filter: ["br"],
+  replacement: () => {
+    return "<p><br></p>";
```
Comment (CarinaWolli) [inline packages/lib/turndownService.ts:8]: allows to add blank lines

**Comment by CarinaWolli** [comment]
PR is ready for review again @zomars The user and team bio is now saved as markdown instead of HTML

File: packages/ui/components/editor/Editor.tsx [hunk 8]
```diff
@@ -12,6 +12,8 @@ import { RichTextPlugin } from "@lexical/react/LexicalRichTextPlugin";
 import { HeadingNode, QuoteNode } from "@lexical/rich-text";
 import { TableCellNode, TableNode, TableRowNode } from "@lexical/table";
 
+import { classNames } from "@calcom/lib";
+
```
Comment (zomars) [inline packages/ui/components/editor/Editor.tsx:16]:
```suggestion
```

File: apps/web/playwright/onboarding.e2e.ts [hunk 9]
```diff
@@ -1,8 +1,11 @@
 /* eslint-disable playwright/no-skipped-test */
 import { expect } from "@playwright/test";
+import MarkdownIt from "markdown-it";
 
 import { test } from "./lib/fixtures";
 
+const md = new MarkdownIt("default", { html: true, breaks: true });
```
Comment (zomars) [inline apps/web/playwright/onboarding.e2e.ts:7]: Unused

**Review by zomars — COMMENTED** [review]
Looking good @CarinaWolli just some nitpicks and orphan imports to merge

**Review by zomars — APPROVED** [review]
LGTM

**Comment by shirazdole** [comment]
very nice!

**Comment by PeerRich** [comment]
@shirazdole casually replying to a 3 week old PR

**Pull request merged**